### PR TITLE
fix: Address pandas 1.5.1 warning with generic assignment

### PR DESCRIPTION
**Pull request by Christos-Hadjinikolis**
## Overview

Chore work here, upgrading some dev libs and getting rid of an irritating new warning from pandas: 
```shell
FutureWarning: In a future version, `df.iloc[:, i] = newvals` will attempt to set the values inplace instead of always setting a new array. To retain the old behavior, use either `df[df.columns[i]] = newvals` or, if columns are non-unique, `df.isetitem(i, newvals)`
  df.loc[:, column_name] = df[column_name].astype(self.schema[column_name])
```

## Checklist

The following won't apply in all cases - mark `N/A` next to point if not needed.

- [x] Appropriate unit tests added/updated
- [x] Module, function, class docstrings updated
- [x] Comments added to PR where necessary
- [x] Interface documentation updated
- [x] Semantic commits used for this PR, so that a CHANGELOG can be generated from them (don't delete your local branch! when tagging a new release from master)

## Release-Steps

- [ ] Merge PR
- [ ] Release new tag

### CODE REVIEW

File: .coveragerc [hunk 1]
```diff
@@ -4,4 +4,4 @@ omit =
     *__init__*
 
 [report]
-fail_under = 94
+fail_under = 93
```
Comment (Christos-Hadjinikolis) [inline .coveragerc:7]: Sorry

File: docs/core.html [hunk 2]
```diff
@@ -288,7 +288,7 @@ <h1 class="title">Module <code>dynamicio.core</code></h1>
                     if len(set([type(v) for v in df[column_name].values])) &gt; 1:  # pylint: disable=consider-using-set-comprehension
                         logger.warning(CASTING_WARNING_MSG.format(column_name, expected_dtype, found_dtype))  # pylint: disable=logging-format-interpolation
                         logger.info(NOTICE_MSG.format(column_name))  # pylint: disable=logging-format-interpolation
-                    df.loc[:, column_name] = df[column_name].astype(self.schema[column_name])
+                    df[column_name] = df[column_name].astype(self.schema[column_name])
```
Comment (maxhipperson) [inline docs/core.html:291]: Just use`pd.assign` ([docs](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.assign.html))
You'll stop getting the warning and the behaviour isn't likely to change.

File: docs/core.html [hunk 3]
```diff
@@ -607,7 +607,7 @@ <h2 id="args">Args</h2>
...
```
Comment (maxhipperson) [inline docs/core.html:610]: As above

File: dynamicio/core.py [hunk 4]
```diff
@@ -259,7 +259,7 @@ def _has_valid_dtypes(self, df: pd.DataFrame) -> bool:
                     if len(set([type(v) for v in df[column_name].values])) > 1:  # pylint: disable=consider-using-set-comprehension
                         logger.warning(CASTING_WARNING_MSG.format(column_name, expected_dtype, found_dtype))  # pylint: disable=logging-format-interpolation
                         logger.info(NOTICE_MSG.format(column_name))  # pylint: disable=logging-format-interpolation
-                    df.loc[:, column_name] = df[column_name].astype(self.schema[column_name])
+                    df[column_name] = df[column_name].astype(self.schema[column_name])
```
Comment (maxhipperson) [inline dynamicio/core.py:262]: Ditto

**Review by maxhipperson — CHANGES_REQUESTED** [review]

File: docs/core.html (hunk 2 above)
Comment (Christos-Hadjinikolis) [reply docs/core.html:291]: > If you do not wish to maintain the original values, then clearly df["A"] = 1 will be more appropriate. This also explains the speed difference, by necessity .assign must copy the data while [...] does not. 

from https://stackoverflow.com/questions/48177914/why-use-pandas-assign-rather-than-simply-initialize-new-column

I don't want to copy anything.

**Review by asfakianakis — APPROVED** [review]

File: docs/core.html (hunk 2 above)
Comment (maxhipperson) [reply docs/core.html:291]: Fair enough

**Review by maxhipperson — APPROVED** [review]

**Pull request merged**